File: NETMouse - .NET release/Examples/C#/Matrices_11.cs
using ABCNET.Extensions;
using ABCNET.Utils;
using System;

namespace TestProgram
{
    internal static class Program
    {
        public static int Max(this int[,] matrix)
        {
            if (matrix == null)
                throw new ArgumentNullException("matrix");

            int max = int.MinValue;
            bool oddFound = false;
            int rowsCount = matrix.GetLength(0);
            int colsCount = matrix.GetLength(1);
            for (int i = 0; i < rowsCount; i++)
                for (int j = 0; j < colsCount; j++)
                    if ((max < matrix[i, j]) && (matrix[i, j] % 2 != 0))
                    {
                        max = matrix[i, j];
                        oddFound = true;
                    }

            if (!oddFound)
                throw new ArgumentException("matrix");

            return max;
        }

        private static void Main(string[] args)
        {
            Matr.Rand(Base.ReadInteger("N:"), Base.ReadInteger("M:"), -27, 38).Print().Max().Println();
        }
    }
}

File: NETMouse - .NET release/Examples/PascalABC.NET/Matrices_11.pas
{$reference ABCNET.dll}
uses ABCNET.Utils;
uses ABCNET.Extensions;
uses System;

function Max(self: array [,] of integer): integer; extensionmethod;
begin
  if self = nil then
    raise new ArgumentNullException('self');

  result := integer.MinValue;
  var oddFound := false;
  var rowsCount := self.GetLength(0);
  var colsCount := self.GetLength(1);
  for var i := 0 to rowsCount - 1 do
    for var j := 0 to colsCount - 1 do
      if (result < self[i, j]) and (self[i, j] mod 2 <> 0) then
      begin
        result := self[i, j];
        oddFound := true;
      end;

  if not oddFound then
    raise new ArgumentException('self');
end;

begin
  Matr.Rand(Base.ReadInt32('N:'), Base.ReadInt32('M:'), -27, 38).Print().Max().Println();
  Base.ReadChar();
end.